instantout/reservation/manager.go

```diff
@@ -0,0 +1,260 @@
+package reservation
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+
+	reservationrpc "github.com/lightninglabs/loop/swapserverrpc"
+)
+
+// Manager manages the reservation state machines.
+type Manager struct {
+	// cfg contains all the services that the reservation manager needs to
+	// operate.
+	cfg *Config
+
+	// activeReservations contains all the active reservationsFSMs.
+	activeReservations map[ID]*FSM
+
+	// currentHeight stores the currently best known block height.
+	currentHeight int32
+
+	// blockEpochChan receives new block heights.
+	blockEpochChan chan int32
+
+	runCtx context.Context
+
+	sync.Mutex
+}
+
+// NewReservationManager creates a new reservation manager.
+func NewReservationManager(cfg *Config) *Manager {
+	return &Manager{
+		cfg:                cfg,
+		activeReservations: make(map[ID]*FSM),
+		blockEpochChan:     make(chan int32),
+	}
+}
+
+// Run runs the reservation manager.
+func (m *Manager) Run(ctx context.Context, height int32) error {
+	// todo(sputn1ck): recover swaps on startup
+	log.Debugf("Starting reservation manager")
+
+	runCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	m.runCtx = runCtx
+	m.currentHeight = height
```

Comment:
nit: i think we can also avoid storing the current height, could just be a local variable and could be passed to `newReservation` so no need for the mutex either.